client/src/lib/error-telemetry.js

```diff
@@ -15,27 +15,42 @@ class ErrorTelemetry {
     this.options = options;
   }
 
+  authenticate(uuid, token) {
+    logger.info(`Error telemetry: uuid=${uuid}, token=${token}`);
+    this.uuid = uuid;
+    this.token = token;
+    return this;
+  }
+
   /**
-   * (Private) default behaviour for the output where to send data to when the watched logging file
-   * has a modification detected.
-   */
-  callErrorTelemetryService(app, logDataObject) {
+  * (Private) default behaviour for the output where to send data to when the watched logging file
+  * has a modification detected.
+  */
+  callErrorTelemetryService(app, logDataObject, uuid, token) {
 
     const api = app.service('telemetry/error');
 
-    return api.create({
-      log: logDataObject
-    }).then((res) => {
-      logger.info('pushed as '+ res.id);
+    const payload = {
+      log: logDataObject,
+      uuid: uuid
+    };
+    return api.create(
+      payload
+      ,
+      {
+        headers: { Authorization: token }
+      }
+    ).then((res) => {
+      logger.info('pushed as ', res);
     }).catch((res) => {
       logger.error('Failed to push log:', res);
     });
   }
 
   /**
-   * monitoredFile: path to the log file to watch
-   * outputFunction(app,line): the function that will be called on each new line detected
-   */
+  * monitoredFile: path to the log file to watch
+  * outputFunction(app,line): the function that will be called on each new line detected
+  */
   setup(monitoredFile, outputFunction=this.callErrorTelemetryService) {
```

Comment:
What I imagine the answer to my previous question about updating the prototype of `callErrorTelemetryService`, I think this class needs to be refactored a bit to disallow this unnecessary passing of an arbitrary `outputFunction` here. 

I thought this was originally meant as a kind of dependency injection for testing, but that doesn't appear the case.

I think this object should be "sealed up" a bit, and the `outputFunction` is an implementation detail that no callers should bother with.